src/lib/eina/eina_inline_lock_win32.x

```diff
@@ -0,0 +1,749 @@
+/* EINA - EFL data type library
+ * Copyright (C) 2020 Lucas Cavalcante de Sousa
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library;
+ * if not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef EINA_INLINE_LOCK_WIN32_X_
+#define EINA_INLINE_LOCK_WIN32_X_
+
+#ifndef WIN32_LEAN_AND_MEAN
+# define WIN32_LEAN_AND_MEAN
+#endif
+
+#include <Windows.h>
+#include <synchapi.h>
+#undef WIN32_LEAN_AND_MEAN
+
+#include "unimplemented.h"
+
+#include <errno.h>
+
+#include <sys/time.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include <sys/types.h>
+#include <unistd.h>
+
+#ifdef EINA_HAVE_DEBUG_THREADS
+#include <assert.h>
+#include <execinfo.h>
+#define EINA_LOCK_DEBUG_BT_NUM 64
+typedef void (*Eina_Lock_Bt_Func) ();
+
+#include "eina_inlist.h"
+#endif
+
+EAPI void _eina_lock_debug_abort(int err, const char *fn, const volatile void *ptr);
+EAPI void _eina_lock_debug_deadlock(const char *fn, const volatile void *ptr);
+
+#define EINA_LOCK_ABORT_DEBUG(err, fn, ptr) \
+   _eina_lock_debug_abort(err, #fn, ptr)
+#define EINA_LOCK_DEADLOCK_DEBUG(fn, ptr) \
+   _eina_lock_debug_deadlock(#fn, ptr)
+
+/* For cond_timedwait */
+#include <time.h>
+#include <sys/time.h>
+
+#include <eina_error.h>
+
+typedef struct _Eina_Lock Eina_Lock;
+typedef struct _Eina_RWLock Eina_RWLock;
+typedef struct _Eina_Condition Eina_Condition;
+typedef struct _Eina_Barrier Eina_Barrier;
+typedef LPDWORD Eina_TLS;
+typedef HANDLE Eina_Semaphore;
+
+#ifdef EINA_HAVE_WIN32_SPINLOCK
+typedef PCRITICAL_SECTION Eina_Spinlock;
+#else
+typedef Eina_Lock Eina_Spinlock;
+#endif
+
+EAPI void eina_lock_debug(const Eina_Lock *mutex);
+
+/** @privatesection  @{ */
+typedef HANDLE _Eina_Thread;
```

Comment:
why underscore in _Eina_Thread?

Reply:
Because this is for internal use only. As on a Linux build `eina_lock` can't be included before/by `eina_lock` I did this `typedef` to make some Windows types more semantic friendly, including defining `_Eina_Thread` as a `HANDLE`.

Reply:
I do the same, but without underscore, because eina_thread.c use `Eina_Thread`. In my case, this typdef is in eina_thread_{win32,posix}.h. Maybe, this avoid "redefinition type" (?!)

Reply:
@caiosignor may I mark this as resolved?